src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.Tree.cs

```diff
@@ -227,6 +191,31 @@ public static void VisitClosures(Scope scope, Action<Scope, Closure> action)
             }
 
             /// <summary>
+            /// Visit all the closures and return true when the <paramref name="func"/> returns
+            /// true. Otherwise, returns false.
+            /// </summary>
+            public static bool CheckClosures(Scope scope, Func<Scope, Closure, bool> func)
```

Comment:
In `InlineThisOnlyEnvironment` when we see if any `ContainingEnvironmentOpt` is non-null.